command/results_go_test2json.go

```diff
@@ -0,0 +1,364 @@
+	actionSkip   = "skip"
+)
+
+// Golang's JSON result output ('test2json') is a newline delimited list
+// of the following struct, encoded as JSON.
+
+// The following rules APPEAR to hold true of Golang's JSON output:
+// 1. Golang offers no field to distinguish between subsequent runs of a test,
+// such as when setting -test.count to a number greater than 1. However,
+// Golang will not run more than one test of the same name at once, even with a
+// call to t.Parallel(). This means that when you observe a "fail" or "pass"
+// action with the same test name, you know that the next time you see "run",
+// it corresponds to a subsequent iteration.
+// 2. Benchmarks do not have a "run" action
+// 3. The Output field has trailing newlines, even on Windows
+// 4. The file emitted by test2json on Windows uses Unix-style line endings
+// 5. Benchmarks do not provide Elapsed, even on fail, except in the package
+// level result
+// 6. Benchmark output will not match the Test. For example,
+// in the benchmark corpus, the output for BenchmarkSomethingSilent-8 is tagged
+// with Test name BenchmarkSomethingElse-8
+type goTest2JSONTestEvent struct {
+	// Time is the time that the line was processed by by go test. Contrary
+	// to the documentation, this string appears to be an RFC3339Nano
+	// formatted string
+	Time time.Time
+
+	// Action is one of:
+	// run    - the test has started running
+	// pause  - the test has been paused
+	// cont   - the test has continued running
+	// pass   - the test passed
+	// bench  - the benchmark printed log output but did not fail
+	// fail   - the test or benchmark failed
+	// output - the test printed output
+	// Additionally, the following is generated by go 1.10.3 but is undocumented:
+	// skip   - the test was skipped
+	Action string
+
+	// Package optionally specifies the go package path of the test being
+	// run
+	Package string
+
+	// Test specifies the name of the test function being tested. When
+	// go test is reporting benchmarks or package level results, Test is
+	// empty
+	Test string
+
+	// Elapsed is the number of seconds a test took. It is set only when
+	// action is pass or fail for tests, and in the package-level pass/fail
+	// action
+	Elapsed float64 // seconds
+
+	// Output is the line that gotest captured, including trailing newlines
+	// or carriage returns. On Windows, the trailing newlines are
+	// Unix-style
+	Output string
+}
+
+type goTest2JSONCommand struct {
+	Files []string `mapstructure:"files" plugin:"expand"`
+
+	base
+}
+
+type goTest2JSONKey struct {
+	name      string
+	iteration int
+}
+
+func goTest2JSONFactory() Command          { return &goTest2JSONCommand{} }
+func (c *goTest2JSONCommand) Name() string { return "gotest.parse_json" }
+
+func (c *goTest2JSONCommand) ParseParams(params map[string]interface{}) error {
+	if err := mapstructure.Decode(params, c); err != nil {
+		return errors.Wrapf(err, "error decoding '%s' params", c.Name())
+	}
+
+	if len(c.Files) == 0 {
```

Comment:
do we want to provide/support gitignore (or even regular) file matching/expansion? the commands that work on files are sort of devided on the question.